File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length - 1;
        int result = 0;
        while(i <= j){
            result = Math.max(Math.min(height[i],height[j]) * (j-i), result);
            if(height[j] > height[i]){
                i++;
            }
            else{
                j--;
            }
        }
        return result;
        
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int: 
        result = 0
        i = 0
        j = len(height) - 1
        while i < j:
            result = max(result, (j-i)*min(height[i], height[j]))
            if(height[i] >= height[j]):
                j = j - 1
            else:
                i = i + 1
        return result

